### PR TITLE
doc: Document Blockbax integration

### PR DESCRIPTION
**Pull request by blockbax**
#### Summary

Added Blockbax integration guide

#### Notes for Reviewers

I got some errors serving Hugo, but I guess it is probably due to me being on Windows. 🤐 I followed all the guidelines and tested it with hugo separately, so I guess it should work. Let me know if not - then I will try to set it up with make on WSL.

```
Error: Error building site: "~\Repos\blockbax\github\lorawan-stack-docs\doc\content\reference\api\events.md:40:1": failed to render shortcode "events": failed to process shortcode: "~\Repos\blockbax\github\lorawan-stack-docs\doc\themes\the-things-stack\layouts\shortcodes\events.html:7:42": execute of template failed: template: shortcodes/events.html:7:42: executing "shortcodes/events.html" at <jsonify>: wrong number of args for jsonify: want 1 got 2
```

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/integrations/cloud-integrations/blockbax/_index.md
Comment (nejraselimovic) [inline doc/content/integrations/cloud-integrations/blockbax/_index.md:26]: This part is a bit unclear. What is the difference between creating subjects manually and creating a subject type? When you say that manually created subjects need to have external IDs matched to device IDs, do you mean TTS device IDs? And what does the `autoCreateSubjects` option do? Can you elaborate that? Feel free to add another commit to explain this a bit for a reader.

**Review by nejraselimovic — APPROVED** [review]
Hi @blockbax, thank you for your contribution! I added a commit to make the doc compliant to our documentation styling. This looks fine now and I'll approve, but there's one thing we should resolve before merging (see the comment below).

File: doc/content/integrations/cloud-integrations/blockbax/_index.md
Comment (blockbax) [reply doc/content/integrations/cloud-integrations/blockbax/_index.md]: Hi @nejraselimovic, thanks for reviewing our contribution - good feedback! 

In a nutshell a subject type is a blueprint you will always need to create manually. A subject is an instance of this type and can be created for you. For example you need to create the generic model of a building (= subject type), but then you can easily add new buildings (= subjects) manually or use `autoCreateSubjects`.

I added some more generic description about the Blockbax concepts and elaborated on `autoCreateSubjects`. Do you think this clears things up?

File: doc/content/integrations/cloud-integrations/blockbax/_index.md
Comment (adriansmares) [inline doc/content/integrations/cloud-integrations/blockbax/_index.md:84]: I think this should be controlled via headers, and have a checkbox in the webhook template.

Comment (blockbax) [reply doc/content/integrations/cloud-integrations/blockbax/_index.md:84]: @adriansmares Can you elaborate a bit more? You mean to create a custom Webhook Template? I wasn't aware this existed but just found it on your docs [here](https://www.thethingsindustries.com/docs/integrations/webhooks/webhook-templates/).

Comment (nejraselimovic) [reply doc/content/integrations/cloud-integrations/blockbax/_index.md:84]: > @adriansmares Can you elaborate a bit more? You mean to create a custom Webhook Template? I wasn't aware this existed but just found it on your docs [here](https://www.thethingsindustries.com/docs/integrations/webhooks/webhook-templates/).

Yes, you can follow the documentation that you linked to create a Blockbax Webhook template and simplify the process of creating the integration. You can also check our [webhook templates repository](https://github.com/TheThingsNetwork/lorawan-webhook-templates/) for the reference. Once the template is there, this PR would be good to go after slight changes.

Comment (blockbax) [reply doc/content/integrations/cloud-integrations/blockbax/_index.md:84]: Thanks will look into it! Currently the platform does not support passing the `autoCreateSubjects` property so hope it is not a blocker to keep it in the payload like this (we will put it on our backlog to update this in the platform). All the other setup we can make much smoother via a webhook template, so will do that for sure!

Comment (adriansmares) [reply doc/content/integrations/cloud-integrations/blockbax/_index.md:84]: Most users will use payload formatters from the device repository, so they won't be able to fill this property in the payload formatter, so my suggestion is to look into loading it from other sources (headers, request parameters).

Comment (blockbax) [reply doc/content/integrations/cloud-integrations/blockbax/_index.md:84]: I understand, but currently we only support passing it via the payload. I will discuss with the team if we can make it a todo to update this on our end. Hope you understand it will take some time and we can merge it for now and come back to improve on it later. 

For the other part we updated the docs and include a webhook template, thanks for the pointer!

Here is the PR for the webhook template: https://github.com/TheThingsNetwork/lorawan-webhook-templates/pull/50

**Comment by nejraselimovic** [comment]
@blockbax this looks great now. I really like that gif on top. Thanks a lot for your contribution and make sure to keep us updated regarding the discussion above.

**Comment by blockbax** [comment]
Thanks a lot @nejraselimovic - we will keep you updated! 🙏

**Pull request merged**
